libs/patchwheel/patch.go

```diff
@@ -0,0 +1,266 @@
+package patchwheel
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/cli/libs/log"
+)
+
+const (
+	versionKey = "Version:"
+	nameKey    = "Name:"
+)
+
+func findMetadataAndRecord(r *zip.ReadCloser, oldDistInfoPrefix string) (metadataFile, recordFile *zip.File) {
+	for _, f := range r.File {
+		if metadataFile == nil && f.Name == oldDistInfoPrefix+"METADATA" {
+			metadataFile = f
+		}
+
+		if recordFile == nil && f.Name == oldDistInfoPrefix+"RECORD" {
+			recordFile = f
+
+			if metadataFile != nil {
+				break
```

Comment:
it won't break early if we find `RECORD` first

Reply:
rewrote this with findFiles(), that does break properly